### PR TITLE
Integrated debug display with dashboard

### PR DESCRIPTION
**Pull request by badeniran**
The command line option is 'd'

### CODE REVIEW

**Comment by enochtsang** [comment]
Can you post a screenshot? And fix the build.

**Comment by badeniran** [comment]
screenshot 16

File: ViewLayer/OverlordWidget/OverlordWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+#include "MPPTUi/I_MpptUi.h"
+#include "TabUi/I_TabUi.h"
+OverlordWidget::OverlordWidget(I_BatteryUi& batteryUi
+                               , I_ControlUi& controlUi
+                               , I_HomePageUi& homepageUi
+                               , I_MotorFaultUi& motorFaultUi
+                               , I_MotorUi& motorUi
+                               , I_MpptUi& mpptUi
+                               , I_TabUi& tabUi
+                              )
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.cpp]: Only initialisation lists should be formatted like this. The arguments should be formatted like:
`OverlordWidget::OverlordWidget(I_BatteryUI& batteryUi,`
`                I_ControlUi& controlUi,`
`...etc`

See the way `DashboardDisplayView.cpp` has it formatted.

File: ViewLayer/OverlordWidget/OverlordWidget.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    connect(&tabUi_.controlButton(), SIGNAL(clicked()),
+            this, SLOT(handleControlButtonClicked()));
+    connect(&tabUi_.motorButton(), SIGNAL(clicked()),
+            this, SLOT(handleMotorButtonClicked()));
+    connect(&tabUi_.motorFaultButton(), SIGNAL(clicked()),
+            this, SLOT(handleMotorFaultButtonClicked()));
+    connect(&tabUi_.mpptButton(), SIGNAL(clicked()),
+            this, SLOT(handleMPPTButtonClicked()));
+    resize(683, 768);
+    QVBoxLayout*    overlordLayout = new QVBoxLayout;
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.cpp]: Extra whitespace between `*` and `overlordLayout`

File: ViewLayer/OverlordWidget/OverlordWidget.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+void OverlordWidget::handleControlButtonClicked()
+{
+    if (!tabUi_.isVisible())
+    {
+        tabUi_.show();
+    }
+
+    menu_->setCurrentWidget(&controlUi_);
+}
+void OverlordWidget::handleHomepageButtonClicked()
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.cpp]: As above.

File: ViewLayer/OverlordWidget/OverlordWidget.cpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+void OverlordWidget::handleMotorFaultButtonClicked()
+{
+    if (!tabUi_.isVisible())
+    {
+        tabUi_.show();
+    }
+
+    menu_->setCurrentWidget(&motorFaultUi_);
+}
+void OverlordWidget::handleMotorButtonClicked()
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.cpp]: As Above.

File: ViewLayer/OverlordWidget/OverlordWidget.cpp [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+{
+    if (!tabUi_.isVisible())
+    {
+        tabUi_.show();
+    }
+
+    menu_->setCurrentWidget(&motorUi_);
+}
+void OverlordWidget::handleMPPTButtonClicked()
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.cpp]: As above.

File: ViewLayer/OverlordWidget/OverlordWidget.h [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+public:
+    explicit OverlordWidget( I_BatteryUi& batteryUi
+                             , I_ControlUi& controlUi
+                             , I_HomePageUi& homepageUi
+                             , I_MotorFaultUi& motorFaultUi
+                             , I_MotorUi& motorUi
+                             , I_MpptUi& mpptUi
+                             , I_TabUi& tabUi
+                           );
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.h]: See my comment in the cpp file on the formatting.

File: ViewLayer/OverlordWidget/OverlordWidget.cpp [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+void OverlordWidget::handleBatteryButtonClicked()
+{
+    if (!tabUi_.isVisible())
+    {
+        tabUi_.show();
+    }
+
+    menu_->setCurrentWidget(&batteryUi_);
+}
```
Comment (bill-luu) [inline ViewLayer/OverlordWidget/OverlordWidget.cpp]: Lets put a new line between the `}` of the previous function and the start of the next.

File: ViewLayer/BatteryUi/BatteryUi.cpp [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#include "BatteryUi.h"
+#include "ui_BatteryUi.h"
+BatteryUi::BatteryUi() :
+    ui_(new Ui::BatteryUi)
+{
+    ui_->setupUi(this);
+}
```
Comment (bill-luu) [inline ViewLayer/BatteryUi/BatteryUi.cpp]: Lets put a new line between the `}` of the previous function and the start of the next, for this and the other source files in this pr.

**Comment by enochtsang** [comment]
Please fix the build.

**Comment by enochtsang** [comment]
Fix merge conflicts

**Comment by enochtsang** [comment]
Fix merge conflicts

**Pull request merged**